### DISCUSSION TITLE
v27 29.10.2021

### DISCUSSION THREAD

**Discussion by dosinabox** [dosinabox/g2nr_unofficial_update#384] [Announcements]
- Два рыцаря около трона Гаронда занимают правильные точки.
- Параметры яркости, контраста и гаммы регулируются плавнее и точнее.
- Если Брайан ушел на маяк, то запись о торговле с ним добавится в раздел дневника 'Торговля во внешнем мире'.
- Хакон не упомянет работу Гарада на паладинов, если ГГ впервые заговорит с ним после первой главы.
- Если Ходжес погиб до третьей главы, то Ларес и Беннет не упомянут его как свидетеля нападения на Лотара.
- Исправлена возможная потеря опыта за диалог с Константино о травах по совету Лестера.
- Исправлено отсутствие оружия у трактирщика на площади правосудия.
- Ларес не пойдет к городу, если ГГ попросит проводить его к магам Воды, находясь у таверны.
- Исправлен перевод реплики ГГ с вопросом о куске руды Кавалорна.
- Исправлена невозможность использования Милтеном, Лестером и Диего дополнительных распорядков дня на Ирдорате.
- Милтен, Лестер и Диего не перезагружают свой распорядок дня при каждом финальном диалоге с ГГ на Ирдорате.
- Заблокирована возможность получения зелья ловкости и второго Глаза Инноса на Ирдорате путем превращения в овцу.
- Нельзя использовать мантии ищущих, полученные с одержимых с помощью опции SystemPack ShowArmor=1.
- Орлан не будет покидать таверну для квеста 'Пей до дна', если церемония принятия ГГ в 'Кольцо Воды' уже началась.
- Мартин не будет забирать у ГГ шпаги бандитов при каждом перечислении улик - для доказательства достаточно одной.
- Дневник Корнелиуса появится в его инвентаре только в третьей главе.
- Все маги Воды за исключением Ватраса и Сатураса теперь носят легкие мантии магов Воды, как в Готике 1.
- Исправлена обратная анимация бокового стрейфа влево с оружием ближнего боя (спасибо ElderGamer).
- Исправлены многочисленные проблемы анимаций при прицеливании луком (спасибо ElderGamer).
- Улучшено позиционирование бутылок при анимации питья (спасибо ElderGamer).
- ГГ прячет горящий факел при взаимодействии с сундуками, котлами и прочими объектами (спасибо ElderGamer).
- Исправлено отсутствие реплики ГГ в диалоге с Орланом о первой оплате комнаты.
- Исправлена боевая музыка при входе в дома Эрола и в лагере Декстера (только с Union, спасибо Gratt).
- Летающие создания перемещаются по поверхности неглубокой воды, а не по дну (только с Union, спасибо Slavemaster).
- Летающие создания будут реже зависать над головой ГГ (только с Union, спасибо Slavemaster).
- Гоблины больше не будут падать при спуске по наклонным поверхностям (спасибо Slavemaster).
- Восстановлены различные текстуры зубов из Готики 1 для людей и зомби.
- Задействованы неиспользуемые анимации взлома и открытия сундуков с замками (только с Union, спасибо Slavemaster).
- Взломанные и открытые замки сундуков будут отображаться как открытые (только с Union, спасибо Slavemaster).
- Исправлено отсутствие текстуры шляпы Грега.
- В диалоге о корабле Халвор не будет пугать паладинами ГГ-паладина.
- Вернувшийся на ферму Бенгара Пардос уйдет в горы вместе с Малаком в третьей главе.
- После диалога о замене Лареса Бальтрам не будет разговаривать с ГГ, одетым в доспехи 'Кольца Воды'.
- Если ГГ будет искать замену Ларесу в доспехах 'Кольца Воды', то получит от Бальтрама в два раза меньше опыта.
- Призванный блуждающий огонек не будет погибать после 60 секунд состояния атаки.
- С судьи снимается бессмертие, что позволило задействовать три неиспользуемые реплики Ли на случай его гибели.
- В поисках орнамента Кавалорн появится ближе к ферме Лобарта, если в третьей главе круг камней занят ищущими.
- Исправлена возможность повторного получения опыта за квесты 'Башня' и 'Северный пляж' у Грега.
- Квест 'Башня' не начнется заново в диалоге с Генри, если он был начат и выполнен у Грега.
- Квест 'Северный пляж' не начнется заново в диалоге с Морганом, если он был начат и выполнен у Грега.
- Исправлена возможность взять с собой пирата у частокола для зачистки башни без разрешения Генри.
- Пираты у частокола не перезагружают свой распорядок дня при завершении квеста 'Башня' у Генри.
- Пираты у частокола не останутся соратниками ГГ при завершении квеста 'Башня' у Грега.
- Диалоги с ищущими с последующей атакой завершаются автоматически.
- Нелегальное проникновение в верхний квартал не блокирует все диалоги с Лотаром.
- После изучения всех мечей у Беннета он будет произносить соответствующую реплику вместо показа пустого диалога.
- Сильвио будет комментировать драки с его людьми независимо от квеста Джарвиса, а также учитывать их убийства.
- Исправлены ситуации, когда давно убитые люди Сильвио не учитывались при перечислении у Джарвиса.
- При первой встрече Ли не назовет новичком ГГ-нарушителя какой-либо гильдии.
- Ли не прокомментирует вступление ГГ в какую-либо гильдию, если ранее не встречал его как нейтрала.
- Запись о торговле у Ибрагима добавляется в дневник также при его упоминании у Фени и горожан.
- При перечислении учителей и торговцев Кардиф не будет упоминать мертвых Карла, Альрика, Игнаца и Ибрагима.
- После бегства в портовый квартал Гербрандт и его жена перестают жаловаться ополчению на преступления ГГ.
- ГГ получит правильное количество опыта за передачу статуэтки Педро.
- Квест 'Хороший боец' перейдет в выполненные сразу после получения одобрения от Корда.
- Исправлено отсутствие реплик людей при пробуждении от магического сна (спасибо Black_Phantom).
- Исправлена бесконечная запись в дневник при вопросе Корнелиусу 'Сколько ты хочешь?'.
- Исправлены нелогичные диалоги о штрафах при уменьшении их количества или после полного прощения грехов.
- Исправлены нелогичные диалоги у Хагена при оплате штрафов у Андрэ и наоборот.
- У кровавых мух не проигрывается несуществующая анимация сна и пробуждения.
- Мертвый Фиск не телепортируется в таверну после диалога со Снафом по наводке от Хуно.
- Если Бифф потеряет ГГ-соратника в Рудниковой долине, то будет ждать его на своей стартовой позиции.
- Исправлена невозможность позвать Биффа с позиций ожидания около болотного, огненного и ледяного драконов.
- Если ГГ не сказал Ксардасу 'Мне нужно оружие', то не теряется возможность спросить его о снаряжении.
- Из торгового инвентаря Фиска не удаляются жала кровавых мух.
- Если Билгот потеряет ведущего его ГГ, то окажется на стартовой позиции у шахты, а не около Прохода.
- Ангар расскажет о своем пропавшем амулете в Хоринисе, даже если не встречал ГГ ранее в Рудниковой долине.
- Исправлена невозможность отдать Ангару его амулет в Хоринисе и на Ирдорате, не узнав о нем в Рудниковой долине.
- С началом второй главы Фернандо перемещается ближе к ратуше, чтобы попросить ГГ об услуге.
- Фернандо восстанавливает свой изначальный распорядок дня, если ГГ уходит в Рудниковую долину не поговорив с ним.
- Благословения Исгарота и Пирокара учитываются Торбеном наравне с благословением Дарона.
- Задействованы две неиспользуемые реплики ГГ о поиске предателей Эстебана.
- Нельзя спровоцировать Декстера на драку, подобрав какой-либо предмет у него за спиной.
- Исправлена невозможность завершить квест 'Отнести репу Хильде', если дать Хильде менее 20 реп.
- Исправлен конфликт диалогов Мо о входе в портовый кабак и о долге Лемару.
- Исправлена блокировка одного из диалогов Лемара, если ГГ закроет свой долг с помощью Лареса.
- Если Василий согласился на 2 золотых за старые монеты, то повторный диалог с торгом не начинается с 1 золотых.
- Нельзя спросить Нагура о его хозяевах, если он не дал ГГ работу и не упомянул их после продажи товаров.
- Исправлена блокировка карманной кражи у Нагура в тюрьме.

<hr /><em>This discussion was created from the release <a href='https://github.com/dosinabox/g2nr_unofficial_update/releases/tag/v27'>v27 29.10.2021</a>.</em>

**Comment by Kor-Angar** [comment]
> Исправлена блокировка карманной кражи у Нагура в тюрьме.

а вообще правильно ли давать возможность красть у НПС в тюрьме учитывая что у них там вроде как ничего нет? :-)

**Reply by dosinabox** [reply]
Если заключенных бить, то предметы у них есть, в том числе ценные

**Comment by Kor-Angar** [comment]
а полное название табличек и свитков в мире отменила?

**Reply by dosinabox** [reply]
В Gothic.ini:
`NameToDescCats=DOCS|MAGIC`

**Comment by Kor-Angar** [comment]
Эрол как то необычно сидел

![20211110152832_1](https://user-images.githubusercontent.com/49514764/141165019-ca026616-7d2d-4616-8218-e334247e3fb3.jpg)

**Reply by dosinabox** [reply]
Не менялся. Бывает.

**Comment by Kor-Angar** [comment]
во второй главе в Минентале / Яркендаре надо что нибудь потестить?

**Reply by dosinabox** [reply]
Если играешь со сборкой где менялись парковки Диего, то в Яркендаре нужно проверить Фиска и Сеньяна:
- Исправлена невозможность начать квест 'Пакет с отмычками', если не спросить Фиска о нападении до встречи в таверне.
- Исправлены нелогичные диалоги и сброс местоположения Фиска, если его первая встреча с ГГ состоялась в таверне.
- Реализована реакция Сеньяна на убийство Эстебана и задействована его неиспользуемая реплика.
- Если ГГ не говорил с Сеньяном, то Эмилио не скажет, что слышал их разговор.
- Диалог с Эмилио об убийстве Сеньяна не добавит запись в неактивный квест 'Покушение на убийство'.

Также следует учесть, что пакет с отмычками был переименован из ItMi_Addon_Lennar_Paket в ItMi_Addon_Fisk_Paket и на старых сохраненках 27 версии его не будет, вставишь читом.

**Reply by dosinabox** [reply]
Видимо, когда-то этот квест давал Леннар)
```cpp
const int Lennar_picklock_amount = 12;
var int LennarPaket_Open;
var int MIS_Lennar_Lockpick;
const int XP_Addon_LennarPaket = 200;
instance ItMi_Addon_Lennar_Paket(C_Item)
func void Use_LennarPaket()
```

**Comment by Kor-Angar** [comment]
Ой, а можно ещё таблички поправить, которые герой с Равена снимает, а то они  квестовые:-(

**Reply by dosinabox** [reply]
А что нужно сделать?

**Reply by Kor-Angar** [reply]
можно попробовать во время разговора равена и ГГ, снять с табличек, которые у Равена флаги "для миссии".

**Reply by Kor-Angar** [reply]
https://github.com/dosinabox/g2nr_unofficial_update/issues/266

**Reply by Kor-Angar** [reply]
> во время разговора равена и ГГ

в DIA_Addon_Raven_Hi_Info или в B_DragonKillCounter

**Comment by Kor-Angar** [comment]
кстати, а вот это 

```D
	if(RavenIsDead == FALSE)
	{
		if(Hlp_GetInstanceID(current_dragon) == Hlp_GetInstanceID(Raven))
		{
			PlayVideoEx("EXTRO_RAVEN.BIK",TRUE,FALSE);
			RavenIsDead = TRUE;
			Saturas_KnowsHow2GetInTempel = TRUE;
			B_RemoveNpc(KDW_14030_Addon_Myxir_ADW);
			B_CheckLog();
		};
	};
```
нелогичнее делать в прекрасной B_CheckDeadMissionNPCs ?

**Reply by dosinabox** [reply]
> нелогичнее делать в прекрасной B_CheckDeadMissionNPCs ?

Надежнее так как сейчас.

**Reply by Kor-Angar** [reply]
так ведь и B_CheckDeadMissionNPCs  и B_DragonKillCounter.d вызываются из ZS_Dead.d

**Reply by dosinabox** [reply]
B_DragonKillCounter() вызывается из цикла ZS_Dead_Loop. В классике с этим уже был проблемы, в НВ сделали лучше и сейчас всё хорошо: https://worldofplayers.ru/threads/41796/post-1072099

**Reply by Kor-Angar** [reply]
тогда из этого следует, что B_CheckDeadMissionNPCs  надо перенести в ZS_Dead_Loop? :-)
